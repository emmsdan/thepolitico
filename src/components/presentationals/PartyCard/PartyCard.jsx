import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './PartyCard.scss';
import Image from '../Image/Image';
const PartyCard = ({ id, logourl, name, hqaddress }) => {
  return (
    <div className="party-card">
      <Link to={`/party/${id}/${name}`}>
        <Image src={`${process.env.IMAGE_CLOUD}/${logourl}`} className="" />
        <div className="info">
          <h2>{name}</h2>
          {hqaddress}
        </div>
      </Link>
    </div>
  );
};

PartyCard.propTypes = {
  name: PropTypes.string,
  hqaddress: PropTypes.string,
  id: PropTypes.string,
  logourl: PropTypes.string,
};

export default PartyCard;
