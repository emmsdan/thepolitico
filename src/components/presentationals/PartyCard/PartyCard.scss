.party-card-1, .party-card-2{
  width: 25%;
  height: 340px;
  margin: 10px;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, .7);
  position: relative;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transform-origin: 100% 0%;
  -moz-transform-origin: 100% 0%;
  -ms-transform-origin: 100% 0%;
  transform-origin: 100% 0%;
  -webkit-transform-style: preserve-3d;
  -moz-transform-style: preserve-3d;
  transform-style: preserve-3d;
  transition: .8s ease-in-out;

  img {
    height: 100%;
    width: 100%;
    cursor: pointer;
    box-shadow: none;
    opacity: .03;
  }
}
.party-card-2 {
  width: 50%;
}
.party-card-1 .back, .party-card-2 .back {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: transparent;
    font-family: 'Arimo', sans-serif;
  }

  .party-card-1 h1, .party-card-2 h1 {
    color: rgb(12, 90, 47);
    text-transform: uppercase;
    font-weight: 400;
    line-height: 1;
    margin-top: 110px;
    text-align: center;
    font-size: 35px;
  }
  .party-card-1 h1 span, .party-card-2 h1 span {
    color: rgb(8, 1, 6);
    display: block;
    margin: 5px;
    font-size: .45em;
    letter-spacing: 3px;
  }
  .party-card-1 h1 i, .party-card-2 h1 i {
    font-style: normal;
    text-transform: none;
    font-family: 'Playfair Display', serif;
  }
  @media only screen and (max-width: 48rem) {
    .party-card-1, .party-card-2 {
      width: 100%;
    }
  }
