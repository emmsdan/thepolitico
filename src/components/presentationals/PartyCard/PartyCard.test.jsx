import React from 'react';
import { shallow } from 'enzyme';

import PartyCard from './PartyCard';

const party = {
  id: 23,
  logourl: 'dfsnsdfds',
  name: 'Daniel',
  hqaddress: 'lekki phase 11',
  className: 'party-card-2',
};
describe('<PartyCard />', () => {
  it('should render correctly', () => {
    const component = shallow(<PartyCard {...party} />);
    expect(component).toMatchSnapshot();
  });
});
