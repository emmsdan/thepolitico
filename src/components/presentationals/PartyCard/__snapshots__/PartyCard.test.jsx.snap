// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartyCard /> should render correctly 1`] = `
<div
  className="party-card-2"
>
  <Link
    to="/party/23/Daniel"
  >
    <Image
      className=""
      src="undefined/dfsnsdfds"
    />
    <div
      className="back"
    >
      <div>
        <br />
        <Image
          src="undefined/dfsnsdfds"
          style={
            Object {
              "boxShadow": "0 1px 2px #000",
              "height": "150px",
              "left": "34%",
              "opacity": 1,
              "position": "absolute",
              "width": "150px",
            }
          }
        />
        <br />
        <br />
        <br />
      </div>
      <h1>
        Daniel
        <span>
          lekki phase 11
        </span>
      </h1>
    </div>
  </Link>
</div>
`;
